### ISSUE TITLE
Talos Network Bonding with Canal CNI is not working when there are two physical interfaces

### ISSUE THREAD

**Issue by pawanmuthya** [siderolabs/talos#5367]
Hi All,
Our clusters have 2 physical interfaces [public and private]. Because of this we are seeing Talos network bonding with Canal CNI hanging and it switches off the pod-pod communication. 

When we tested in one cluster without the public interface, the K8S internal communications were working totally fine. But once the public interface is enabled the K8S takes the public interface as the default interface and it starts to fail. The moment it's removed, Canal CNI keeps dying saying the default network interface is not available. 

Any tips or answers here would be much appreciated. 

Thanks.

**Comment by smira** [comment]
I'm not sure about all the details, but have you got kubelet IP fixed to use the private network using the machine configuration?

https://www.talos.dev/v1.0/reference/configuration/#kubeletnodeipconfig

```yaml
machine:
  kubelet:
    nodeIP:
      validSubnets:
           - ["10.0.0.0/8"]
```

From your question, it's not clear how interface bonding is related to the problem.

**Comment by pawanmuthya** [comment]
Hi,
The Talos cluster I'm using is v0.11.4. Can you please let me know how to configure a fixed IP for kubelet in v0.11.4. I scanned through the documents but couldn't figure it out.
Is this fixed IP for kubelet something that has been introduced only in v1.0.0+ versions? 
Thanks.

**Comment by smira** [comment]
> fixed ip for kubelet was introduced in 0.14 if my memory serves me right

with Talos 0.11 it's still possible to set the IP manually via `machine.kubelet.extraArgs`: https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/ `--node-ip=10.5.0.3`
